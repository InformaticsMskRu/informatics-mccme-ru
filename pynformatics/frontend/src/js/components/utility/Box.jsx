import React from 'react';
import styled from 'styled-components';
import { palette } from 'styled-theme';

import { BoxWrapper as IsoBoxWrapper } from '../../isomorphic/components/utility/box.style';
import {
  BoxTitle as IsoBoxTitleWrapper,
  BoxSubTitle as IsoBoxSubTitleWrapper
} from '../../isomorphic/components/utility/boxTitle.style';


const BoxWrapper = IsoBoxWrapper.extend`
  border-radius: 4px;
  color: ${palette("other", 7)};
  
  &&& {
    margin: 0;
  }
`;

const BoxTitleWrapper = IsoBoxTitleWrapper.extend`
  font-size: 22px;
  margin-right: 10px;
  display: inline-block;
  margin-bottom: 0;
  color: ${palette("other", 7)};
`;

const BoxSubTitleWrapper = IsoBoxSubTitleWrapper.extend`
  font-size: 14px;
  display: inline-block;
  margin-bottom: 0;
`;

const BoxHeaderWrapper = styled.div`
  margin-bottom: 20px;
`;

const BoxHeader = ({title, subtitle}) => (
  <BoxHeaderWrapper>
    { title ? <BoxTitleWrapper>{' '}{title}{' '}</BoxTitleWrapper> : '' }
    { subtitle ? <BoxSubTitleWrapper>{' '}{subtitle}{' '}</BoxSubTitleWrapper>: '' }
  </BoxHeaderWrapper>
);

const Box = ({ title, subtitle, children, style, className }) => (
  <BoxWrapper style={style} className={className}>
    { title || subtitle ? <BoxHeader title={title} subtitle={subtitle}/> : null }
    { children }
  </BoxWrapper>
);

export default Box;
