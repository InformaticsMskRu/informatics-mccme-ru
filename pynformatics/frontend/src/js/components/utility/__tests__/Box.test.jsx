import React from 'react';
import renderer from 'react-test-renderer';
import 'jest-styled-components';

import Box from '../Box';


describe('renders correctly', () => {
  it('when empty', () => {
    const tree = renderer.create(
      <Box />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with title', () => {
    const tree = renderer.create(
      <Box title="Some title"/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with subtitle', () => {
    const tree = renderer.create(
      <Box subtitle="Some subtitle"/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with title and subtitle', () => {
    const tree = renderer.create(
      <Box title="Some title" subtitle="Some subtitle"/>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('with children', () => {
    const tree = renderer.create(
      <Box title="Some title" subtitle="Some subtitle">
        <div>child 1</div>
        <div>child 2</div>
      </Box>
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
