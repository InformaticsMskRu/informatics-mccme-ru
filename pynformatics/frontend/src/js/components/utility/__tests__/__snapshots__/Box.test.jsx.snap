// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when empty 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 20px;
  background-color: #ffffff;
  border: 1px solid;
  margin: 0 17px 30px;
  border-radius: 4px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0.half {
  width: calc(50% - 34px);
}

.c0.c0.c0 {
  margin: 0;
}

@media only screen and (max-width:767px) {
  .c0 {
    padding: 20px;
    margin: 0 10px 30px;
  }
}

@media (max-width:767px) {
  .c0.half {
    width: 100%;
  }
}

<div
  className="c0"
  style={undefined}
/>
`;

exports[`renders correctly with children 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 20px;
  background-color: #ffffff;
  border: 1px solid;
  margin: 0 17px 30px;
  border-radius: 4px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0.half {
  width: calc(50% - 34px);
}

.c0.c0.c0 {
  margin: 0;
}

.c2 {
  font-size: 14px;
  font-weight: 500;
  margin: 0;
  margin-bottom: 5px;
  font-size: 22px;
  margin-right: 10px;
  display: inline-block;
  margin-bottom: 0;
}

.c3 {
  font-size: 13px;
  font-weight: 400;
  line-height: 24px;
  font-size: 14px;
  display: inline-block;
  margin-bottom: 0;
}

.c1 {
  margin-bottom: 20px;
}

@media only screen and (max-width:767px) {
  .c0 {
    padding: 20px;
    margin: 0 10px 30px;
  }
}

@media (max-width:767px) {
  .c0.half {
    width: 100%;
  }
}

<div
  className="c0"
  style={undefined}
>
  <div
    className="c1"
  >
    <h3
      className="c2"
    >
       
      Some title
       
    </h3>
    <p
      className="c3"
    >
       
      Some subtitle
       
    </p>
  </div>
  <div>
    child 1
  </div>
  <div>
    child 2
  </div>
</div>
`;

exports[`renders correctly with subtitle 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 20px;
  background-color: #ffffff;
  border: 1px solid;
  margin: 0 17px 30px;
  border-radius: 4px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0.half {
  width: calc(50% - 34px);
}

.c0.c0.c0 {
  margin: 0;
}

.c2 {
  font-size: 13px;
  font-weight: 400;
  line-height: 24px;
  font-size: 14px;
  display: inline-block;
  margin-bottom: 0;
}

.c1 {
  margin-bottom: 20px;
}

@media only screen and (max-width:767px) {
  .c0 {
    padding: 20px;
    margin: 0 10px 30px;
  }
}

@media (max-width:767px) {
  .c0.half {
    width: 100%;
  }
}

<div
  className="c0"
  style={undefined}
>
  <div
    className="c1"
  >
    
    <p
      className="c2"
    >
       
      Some subtitle
       
    </p>
  </div>
</div>
`;

exports[`renders correctly with title 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 20px;
  background-color: #ffffff;
  border: 1px solid;
  margin: 0 17px 30px;
  border-radius: 4px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0.half {
  width: calc(50% - 34px);
}

.c0.c0.c0 {
  margin: 0;
}

.c2 {
  font-size: 14px;
  font-weight: 500;
  margin: 0;
  margin-bottom: 5px;
  font-size: 22px;
  margin-right: 10px;
  display: inline-block;
  margin-bottom: 0;
}

.c1 {
  margin-bottom: 20px;
}

@media only screen and (max-width:767px) {
  .c0 {
    padding: 20px;
    margin: 0 10px 30px;
  }
}

@media (max-width:767px) {
  .c0.half {
    width: 100%;
  }
}

<div
  className="c0"
  style={undefined}
>
  <div
    className="c1"
  >
    <h3
      className="c2"
    >
       
      Some title
       
    </h3>
    
  </div>
</div>
`;

exports[`renders correctly with title and subtitle 1`] = `
.c0 {
  width: 100%;
  height: 100%;
  padding: 20px;
  background-color: #ffffff;
  border: 1px solid;
  margin: 0 17px 30px;
  border-radius: 4px;
}

.c0:last-child {
  margin-bottom: 0;
}

.c0.half {
  width: calc(50% - 34px);
}

.c0.c0.c0 {
  margin: 0;
}

.c2 {
  font-size: 14px;
  font-weight: 500;
  margin: 0;
  margin-bottom: 5px;
  font-size: 22px;
  margin-right: 10px;
  display: inline-block;
  margin-bottom: 0;
}

.c3 {
  font-size: 13px;
  font-weight: 400;
  line-height: 24px;
  font-size: 14px;
  display: inline-block;
  margin-bottom: 0;
}

.c1 {
  margin-bottom: 20px;
}

@media only screen and (max-width:767px) {
  .c0 {
    padding: 20px;
    margin: 0 10px 30px;
  }
}

@media (max-width:767px) {
  .c0.half {
    width: 100%;
  }
}

<div
  className="c0"
  style={undefined}
>
  <div
    className="c1"
  >
    <h3
      className="c2"
    >
       
      Some title
       
    </h3>
    <p
      className="c3"
    >
       
      Some subtitle
       
    </p>
  </div>
</div>
`;
